### ISSUE TITLE
Implement pillar puzzles

### ISSUE THREAD

**Issue by daviddeutsch** [aren227/TheWitness-LockScreen#32]
This is also mentioned in the ToDo list in the README.md, but I thought it might make sense to give it a proper issue.

There is a similar issue for The Windmill: https://github.com/thefifthmatt/windmill-client/issues/17

I found the idea of having it pseudo-3D quite interesting. What were your plans for this? Did you want to make it in 3D?

**Comment by aren227** [comment]
Yes, I want to make it 3D. Also, Like in [witnesspuzzles.com](https://witnesspuzzles.com/pages/browse.html), I'm planning to add a button that enables 2D mode which makes puzzle solving more easier.

But I'm not sure about touch controls in pillar puzzles. If we are using the mouse like the pc version of the witness, We can fix the mouse cursor and rotate a pillar with its horizontal move. But when it comes to mobile, We should think of quite different way.

For now, my plan is that if we move finger slightly to the right from the start point, it will rotate constantly clockwise, and the other way around.

**Comment by daviddeutsch** [comment]
:thinking: hmm. What might be fun is if there was a separate control at the bottom, like a rotation wheel, where you control the rotation with one finger and then with the another one control the line.

That's different to the game, of course, it would be kind of like having to use your left and right keys for horizontal moves while the mouse could do horizontal AND vertical moves, but be restricted by what your left-right moves make available to you.

I think the main aspect to consider here is that your left-right movement is restricted to the cursor precisely so that you cannot "look around" and are a little bit stuck where you are once you started. Especially on the last two Challenge puzzles, that adds to the frenzy because you can just kinda eyeball what the map of the puzzle is and then have to commit to what limited information you could retain about the layout.

I think the most faithful implementation would be what you suggested - that the puzzle tries to accomodate left-right movement by moving the rotation accordingly and is a little forgiving when you want to move your finger back to the middle to make your finger position match your cursor position.

**Comment by daviddeutsch** [comment]
Maybe there could be an additional feature that emulates the ability to "walk around" the pillar where you are able to rotate it without placing a cursor, so kind of just like rotating a virtual cylinder. Then, once you place your cursor, that ability is gone and you're locked to the cursor like in the game.
